types/externs.d.ts

```diff
@@ -120,10 +120,14 @@ declare global {
       gatherMode?: boolean | string;
       /** Flag indicating that the browser storage should not be reset for the audit. */
       disableStorageReset?: boolean;
-      /** How emulation (useragent, device screen metrics, touch) should be applied. `none` indicates Lighthouse should leave the host browser as-is. */
-      emulatedFormFactor?: 'mobile'|'desktop'|'none';
-      /** Dangerous setting only to be used by Lighthouse team. Disables the device metrics and touch emulation that emulatedFormFactor defines. Details in emulation.js */
-      internalDisableDeviceScreenEmulation?: boolean
+
+      // TODO: formFactor should really be required, but since this type is shared in Flags, we can't reasonably do that. We'll likely need to separate SharedFlagsSettings into CLI Flags and Config Settings.
+      /** How Lighthouse should interpret this run in regards to scoring performance metrics and skipping mobile-only tests in desktop. Must be set even if throttling/emulation is being applied outside of Lighthouse. */
+      formFactor?: 'mobile'|'desktop';
+      /** Screen emulation properties (width, height, dpr, mobile viewport) to apply. Typically set to false if mobile emulation is applied outside of Lighthouse, or its being run on a mobile device. For desktop, we recommend applying consistent desktop screen emulation. */
+      screenEmulation?: Pick<LH.Crdp.Emulation.SetDeviceMetricsOverrideRequest, 'width'|'height'|'deviceScaleFactor'|'mobile'> | false;
```

Comment:
feels like the the none option should be `null`, not `false`?

Reply:
When @paulirish and I discussed this we (maybe I?) had a few problems that pushed for `false`, one much more important than the others :)

1. we want a definitive "turn this off" and `false` feels far more explicit in that sense than the interpretation of `null`
2. `null` is the "not set" state for many existing settings due to `key in` problems
3. `false` happens to work naturally with CLI flag semantics

Reply:
> * we want a definitive "turn this off" and `false` feels far more explicit in that sense than the interpretation of `null`

I can understand that, even if a `screenEmulation: null` in a config's settings also seems pretty explicit to me too (and probably more idiomatic js but not a huge deal)

> * `null` is the "not set" state for many existing settings due to `key in` problems

What `key in` problems are there? Shouldn't `merge()` handle those correctly?

> 3\. `false` happens to work naturally with CLI flag semantics

This I'm not sure I agree with :) It works with `--no-` so it's nice that the opposite of `--emulated-user-agent="Mozilla/5.0..."` is `--no-emulated-user-agent`, but if you saw `--no-emulated-user-agent` in the docs, you might of course assume the opposite of that is `--emulated-user-agent` on its own.

Without a type yargs is also pretty wide open, e.g. `--screen-emulation 5` will parse without problem and that `--emulated-user-agent` will happily give `true`, so we'll have to take on the parsing/erroring part of using these flags from the CLI.

Reply:
tsc also often widens `false`s to `boolean`, somewhat even more aggressively than `'literal'` -> `string` because it's probably very rare for functions to accept a `false` without a `true`, so expect some `/** @type {false} */ (false)`s in our future :)

e.g. if you add a `emulatedUserAgent: false` to one of the smokehouse configs in this branch, you'll get a `Type 'boolean' is not assignable to type 'false | string | undefined'.`

Reply:
(admittedly I may care too much about this right now because I'm dealing with annoying changes to yargs, like all the default-handling behavior changing when updating to latest :P)

Reply:
> even if a screenEmulation: null in a config's settings also seems pretty explicit to me too (and probably more idiomatic js but not a huge deal)

Not sure how to settle this then 🤷  I would be surprised if a majority of JS devs say using `null` in an API to mean "off" is more idiomatic than, well than almost anything else really. I consider our limited use of `null` today to be an unfortunate *non*-JS pattern (and in many places a wart of the `key in` issue we would want to remove), especially considering the number of prominent figures these days recommending against using null *anywhere*.

> What key in problems are there? Shouldn't merge() handle those correctly?

It's not *entirely* about merge (though attempting to merge `null` into an object default today will give you problems) it's about cleaning the unrecognized CLI flags from polluting settings in a typechecked way.

https://github.com/GoogleChrome/lighthouse/blob/2cbbd289be33a586462ed248179df7d7b692ca53/lighthouse-core/config/config.js#L187-L197

https://github.com/GoogleChrome/lighthouse/blob/2cbbd289be33a586462ed248179df7d7b692ca53/lighthouse-core/config/constants.js#L67-L77

> so we'll have to take on the parsing/erroring part of using these flags from the CLI.

I'll just use one of your favs here :)

<img src="https://user-images.githubusercontent.com/2301202/101522663-27ec6180-394d-11eb-8e26-faf4e3685d0b.png" width=200>

Reply:
Credit where credit is due. https://github.com/GoogleChrome/lighthouse/pull/11794 made it farther than any parsing/erroring attempt has gone before :)

Reply:
talked with brendan about this and the yargs things.

we ended up with `false`.

Reply:
Yes, I think `null` is a pretty normal `None` in js when `undefined` and optional are off the table, and nullable types _by default_ (with no checked dereferencing) was the billion dollar mistake, not opt-in and checked nullability (which is just an uglier `Option`, which all the prominent figures these days do like) :P

But I was trying to say I can get behind `false` for being explicit and somewhat self-documenting. I found a mostly good solution for the yargs side of things over in [#11794](https://github.com/GoogleChrome/lighthouse/pull/11794/files#diff-9faf90ac891f9184635c19d214e8c030c7ceb61720cba456d8eef764af5b5c25R316-R325) for `-G` and `-A` (`string|boolean`), and we can live with the typescript thing.

Reply:
update: after this discussion i spoke with shane who brought up protobuf compat and indicated the above options aren't great. he suggested `screenEmulation.disabled: boolean` which is what we've gone with